datalad_fuse/fsspec.py

```diff
@@ -0,0 +1,38 @@
+from pathlib import Path
+from typing import IO, Union
+from datalad.support.annexrepo import AnnexRepo
+import fsspec
+from fsspec.implementations.cached import CachingFileSystem
+
+
+class FsspecAdapter:
+    def __init__(self, path: Union[str, Path]) -> None:
+        self.annex = AnnexRepo(str(path))
+        self.cache_dir = Path(path, ".git", "datalad", "cache", "fsspec")
+        self.fs = CachingFileSystem(
+            fs=fsspec.filesystem("http"),
+            # target_protocol='blockcache',
+            cache_storage=str(self.cache_dir),
+            # cache_check=600,
+            # block_size=1024,
+            # check_files=True,
+            # expiry_times=True,
+            # same_names=True
+        )
+
+    def get_url(self, filepath: Union[str, Path]) -> str:
+        whereis = self.annex.whereis(str(filepath), output="full")
+        for v in whereis.values():
+            if v["description"] == "web":
```

Comment:
I said "web", but let's immediately generalize further since one of the most prominent use cases doesn't work

```shell
(git-annex)lena:~/datalad/openneuro/ds000002[master]sub-01/anat
$> datalad fsspec-head -n 1 sub-01_T1w.nii.gz
[ERROR  ] KeyError('whereis') (KeyError) 
(dev3) 1 15660 ->1.....................................:Thu 16 Sep 2021 02:33:41 PM EDT:.
(git-annex)lena:~/datalad/openneuro/ds000002[master]sub-01/anat
$> datalad --version
datalad 0.15.0+2.g1e0f732c1
(dev3) 1 15661.....................................:Thu 16 Sep 2021 02:33:49 PM EDT:.
(git-annex)lena:~/datalad/openneuro/ds000002[master]sub-01/anat
$> git annex whereis sub-01_T1w.nii.gz
whereis sub-01_T1w.nii.gz (2 copies) 
  	504faad4-7ce0-4b16-868e-8c9b8c4e844f -- root@2964ce471527:/datalad/ds000002
   	65f67b65-70f7-4b89-8858-30bb4e367209 -- [s3-PUBLIC]

  s3-PUBLIC: https://s3.amazonaws.com/openneuro.org/ds000002/sub-01/anat/sub-01_T1w.nii.gz?versionId=9aYtnv2csgYcAbRbQgtIiKwkP2tVxFSH
ok
(dev3) 1 15662.....................................:Thu 16 Sep 2021 02:33:57 PM EDT:.
(git-annex)lena:~/datalad/openneuro/ds000002[master]sub-01/anat
$> git remote show -n origin
* remote origin
  Fetch URL: http://datasets.datalad.org/openneuro/ds000002/.git

```

- error suggests need for a fix (may be here - no description?)
- do not bother checking description, just check if url has http or https protocol. (that is what we match for driver in fsspec right?)

Reply:
The method now returns all HTTP(S) URLs.  I don't know what the source of that `KeyError` is, but I don't believe it's from my code.

Reply:
do not observing the KeyError now, so all good, thank you